src/System.Private.CoreLib/shared/System/Array.cs

```diff
@@ -827,14 +481,15 @@ public static int BinarySearch(Array array, int index, int length, object value,
                     // i might overflow if lo and hi are both large positive numbers. 
                     int i = GetMedian(lo, hi);
 
-                    int c = 0;
+                    int c;
                     try
                     {
                         c = comparer.Compare(objArray[i], value);
                     }
                     catch (Exception e)
                     {
                         ThrowHelper.ThrowInvalidOperationException(ExceptionResource.InvalidOperation_IComparerFailed, e);
+                        return default;
```

Comment:
Why are these returns needed?  Do they help the code gen somehow?

Reply:
To help C# compiler flow analysis not to require c double initialization